tests: drivers: can: mcx_n5xx_evk/mcxn547/cpu0 : build failure with clock support

### Describe the bug

- mcx_n5xx_evk/mcxn547/cpu0 

commit a52ff9a7bb64524acbe3ec1326cb2333acb2bcf5
Author: Neil Chen <cheng.chen_1@nxp.com>
Date:   Fri Oct 17 17:49:18 2025 +0800

    drivers: syscon: update syscon driver to support mcxa flexcan clock
    
    Add mcxa flexcan clock support
    
    Signed-off-by: Neil Chen <cheng.chen_1@nxp.com>

 drivers/clock_control/clock_control_mcux_syscon.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

### Regression

- [x] This is a regression.

### Steps to reproduce

```
west build -b mcx_n5xx_evk/mcxn547/cpu0 tests/drivers/can/shell
```

### Relevant log output

```shell
[122/194] Building C object zephyr/drivers/can/CMakeFiles/drivers__can.dir/can_fake.c.obj
[123/194] Building C object zephyr/drivers/can/CMakeFiles/drivers__can.dir/can_mcux_flexcan.c.obj
[124/194] Linking C static library zephyr/drivers/can/libdrivers__can.a
[125/194] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_syscon.c.objFAILED: zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_syscon.c.obj e-promotion -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/tests/drivers/can/shell=CMAKE_SOURCE_DIR -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/home/jenkins/agent/workspace/zephyr_test_pipe=WEST_TOPDIR -ffunction-sections -fdata-sections -mcmse -specs=picolibc.specs -std=c17 -MD -MT zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_syscon.c.obj -MF zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_syscon.c.obj.d -o zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_mcux_syscon.c.obj -c /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/clock_control/clock_control_mcux_syscon.c/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/clock_control/clock_control_mcux_syscon.c: In function 'mcux_lpc_syscon_clock_control_get_subsys_rate':/home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/clock_control/clock_control_mcux_syscon.c:615:25: error: too few arguments to function 'CLOCK_GetFlexcanClkFreq' 615 
| *rate = CLOCK_GetFlexcanClkFreq(); 
| ^~~~~~~~~~~~~~~~~~~~~~~In file included from /home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/common/fsl_common_arm.h:652, from /home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/common/fsl_common.h:353, from /home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk-ng/drivers/port/fsl_port.h:11, from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/soc/nxp/mcx/mcxn/./soc.h:12, from /home/jenkins/agent/workspace/zephyr_test_pipe/zephyr/drivers/clock_control/clock_control_mcux_syscon.c:11:/home/jenkins/agent/workspace/zephyr_test_pipe/modules/hal/nxp/mcux/mcux-sdk-ng/devices/MCX/MCXN/MCXN947/drivers/fsl_clock.h:1851:10: note: declared here 1851 
| uint32_t CLOCK_GetFlexcanClkFreq(uint32_t id); 
| ^~~~~~~~~~~~~~~~~~~~~~~ninja: build stopped: subcommand failed.
```

### Impact

Annoyance – Minor irritation; no significant impact on usability or functionality.

### Environment

- OS: ubuntu
- toolchain: zephyr
- version: v4.3.0-3988-g3dfe5a98f6d9

### Additional Context

_No response_

--- a/drivers/clock_control/clock_control_mcux_syscon.c
+++ b/drivers/clock_control/clock_control_mcux_syscon.c
@@ -610,7 +610,7 @@ static int mcux_lpc_syscon_clock_control_get_subsys_rate(const struct device *de
 
 #if defined(CONFIG_CAN_MCUX_FLEXCAN)
 #if (defined(FSL_FEATURE_SOC_FLEXCAN_COUNT) && (FSL_FEATURE_SOC_FLEXCAN_COUNT == 1) && \
-	!defined(CONFIG_SOC_MCXA346))
+	!defined(CONFIG_SOC_MCXA346) && !defined(CONFIG_SOC_MCXN547))
 	case MCUX_FLEXCAN0_CLK:
 		*rate = CLOCK_GetFlexcanClkFreq();
 		break;

